src/util/async/README.md

```diff
@@ -0,0 +1,231 @@
+# Async framework
+
+## Introduction
+
+Clio uses threads intensively. Multiple parts of Clio were/are implemented by running a `std::thread` with some sort of loop inside. Every time this pattern is reimplemented in a slightly different way. State is managed using asynchronous queues, atomic flags, mutexes and other low level primitives.
+
+On the other hand, Clio also uses `Boost.Asio` for more complex tasks such as networking, scheduling RPC handlers, and even interacting with the database is done via Asio’s coroutines.
+
+There was a need for a simple yet powerful framework that will cover the following in a unified way:
```

Comment:
in an unified way

Reply:
I believe `a` is correct here. Checked with https://www.grammarly.com/sentence-checker as well 👍